frontend/src/Editor/DraggableBox.jsx

```diff
@@ -63,271 +63,287 @@ function getStyles(isDragging, isSelectedComponent) {
-        currentLayout,
-      },
-      collect: (monitor) => ({
-        isDragging: monitor.isDragging(),
+    const [{ isDragging }, drag, preview] = useDrag(
+      () => ({
+        type: ItemTypes.BOX,
+        item: {
+          id,
+          title,
+          component,
+          zoomLevel,
+          parent,
+          layouts,
+          canvasWidth,
+          currentLayout,
+        },
+        collect: (monitor) => ({
+          isDragging: monitor.isDragging(),
+        }),
       }),
-    }),
-    [id, title, component, index, zoomLevel, parent, layouts, canvasWidth, currentLayout]
-  );
+      [id, title, component, index, currentLayout, zoomLevel, parent, layouts, canvasWidth]
+    );
 
-  useEffect(() => {
-    preview(getEmptyImage(), { captureDraggingState: true });
-  }, [isDragging]);
+    useEffect(() => {
+      preview(getEmptyImage(), { captureDraggingState: true });
+    }, [isDragging]);
 
-  useEffect(() => {
-    if (resizingStatusChanged) {
-      resizingStatusChanged(isResizing);
-    }
-  }, [isResizing]);
+    useEffect(() => {
+      if (resizingStatusChanged) {
+        resizingStatusChanged(isResizing);
+      }
+    }, [isResizing]);
 
-  useEffect(() => {
-    if (draggingStatusChanged) {
-      draggingStatusChanged(isDragging2);
-    }
+    useEffect(() => {
+      if (draggingStatusChanged) {
+        draggingStatusChanged(isDragging2);
+      }
 
-    if (isDragging2 && !isSelectedComponent) {
-      setSelectedComponent(id, component);
-    }
-  }, [isDragging2]);
+      if (isDragging2 && !isSelectedComponent) {
+        setSelectedComponent(id, component);
+      }
+    }, [isDragging2]);
 
-  const style = {
-    display: 'inline-block',
-    alignItems: 'center',
-    justifyContent: 'center',
-    padding: '0px',
-  };
+    const style = {
+      display: 'inline-block',
+      alignItems: 'center',
+      justifyContent: 'center',
+      padding: '0px',
+    };
 
-  let _refProps = {};
+    let _refProps = {};
 
-  if (mode === 'edit' && canDrag) {
-    _refProps = {
-      ref: drag,
-    };
-  }
+    if (mode === 'edit' && canDrag) {
+      _refProps = {
+        ref: drag,
+      };
+    }
 
-  function changeCanDrag(newState) {
-    setCanDrag(newState);
-  }
+    const changeCanDrag = useCallback(
+      (newState) => {
+        setCanDrag(newState);
+      },
+      [setCanDrag]
+    );
 
-  const defaultData = {
-    top: 100,
-    left: 0,
-    width: 445,
-    height: 500,
-  };
+    const defaultData = {
+      top: 100,
+      left: 0,
+      width: 445,
+      height: 500,
+    };
 
-  const layoutData = inCanvas ? layouts[currentLayout] || defaultData : defaultData;
-  const gridWidth = canvasWidth / NO_OF_GRIDS;
-  const width = (canvasWidth * layoutData.width) / NO_OF_GRIDS;
+    const layoutData = inCanvas ? layouts[currentLayout] || defaultData : defaultData;
+    const gridWidth = canvasWidth / 43;
+    const width = (canvasWidth * layoutData.width) / 43;
```

Comment:
Seems like these got changed while resolving the merge conflicts. Please check and update